Apply store's filter to layer in FeatureStore

chrismayer opened the pull request:
This ensures that a possible filter set to a FeatureStore is applied to the underlying feature layer (if existing). Otherwise there is an inconsistency between the data in the store and the features on the map.

TIA for any review!

marcjansen commented on src/data/store/Features.js:
```diff
@@ -274,6 +276,37 @@ Ext.define('GeoExt.data.store.Features', {
+                // collect the filtered features in the store
+                var filteredFeatures = [];
+                me.each(function(rec) {
+                    filteredFeatures.push(rec.data);
```
Or `rec.getFeature`? Untested.

chrismayer replied:
Unfortunately ``rec.getFeature()`` did not work... I was also a bit surprised.

marcjansen commented on src/data/store/Features.js:
```diff
@@ -274,6 +276,37 @@ Ext.define('GeoExt.data.store.Features', {
+
+                delete me._filtering;
+
+                me.bindLayerEvents();
```
Should the last two lines be switched?

chrismayer replied:
Yes, would be better.

marcjansen commented:
Hey @chrismayer.

What happens if the original layer that was used to configure the FeatureStore was using a source with an `url`? Won't calling clear on the `source reload` the URL?

You can e.g. test this here in the OpenLayers examples: http://openlayers.org/en/latest/examples/kml-earthquakes.html . 

If you issue

```
vector.getSource().clear();
``` 

the KML is reloaded and the source repopulated.

I think we should at least guard this with a flag like `filterVector: true/false` (open to suggestions wrt the actual name, possibly `passThroughFilter`).

Once this is configurable, I am :+1: for merging.

chrismayer commented:
Thanks for your review @marcjansen!
Valid point with the possible layer-source, which is configured with an URL. I will address this as you suggested by introducing a config option.

chrismayer commented:
So, I added a config flag to switch on/off the filter forwarding to the layer, addressed the minor comments of @marcjansen, added tests and an example.

marcjansen commented:
Very cool, thanks. Looking very good to me, please merge.
